test/widgets/app_test.dart

```diff
@@ -51,4 +54,56 @@ void main() {
       ]);
     });
   });
+
+  group('choose account page', () {
+    Future<void> setupChooseAccountPage(WidgetTester tester, {
+      List<Account>? accounts,
+    }) async {
+      addTearDown(testBinding.reset);
+
+      for (final account in accounts ?? []) {
+        await testBinding.globalStore
+          .insertAccount(account.toCompanion(false));
+      }
+
+      await tester.pumpWidget(
+        const MaterialApp(
+          localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+          supportedLocales: ZulipLocalizations.supportedLocales,
+          home: GlobalStoreWidget(
+            child: ChooseAccountPage())));
+
+      // global store gets loaded
+      await tester.pumpAndSettle();
+    }
+
+    List<Account> dummyAccounts(int count) {
+      return List.generate(count, (i) => eg.account(
+        id: i,
+        user: eg.user(fullName: 'User $i', email: 'user$i@example'),
+        apiKey: 'user${i}apikey',
+      ));
+    }
+
+    Finder findAccount(Account account) => find.text(account.email).hitTestable();
+
+    void checkAccountShown(Account account, {required bool expected}) {
+      check(findAccount(account).evaluate().length).equals(expected ? 1 : 0);
```

Comment:
nit: do more processing on the result of `check`, instead of its argument:
```suggestion
      check(findAccount(account).evaluate()).length.equals(expected ? 1 : 0);
```

This allows `package:checks` to give more informative output if the check fails.